frame/democracy/src/lib.rs

```diff
@@ -0,0 +1,2084 @@
+		/// want to set this permanently to `false`, others may want to condition it on things such
+		/// as an upgrade having happened recently.
+		#[pallet::constant]
+		type InstantAllowed: Get<bool>;
+
+		/// Minimum voting period allowed for a fast-track referendum.
+		#[pallet::constant]
+		type FastTrackVotingPeriod: Get<Self::BlockNumber>;
+
+		/// Slashed stakes are returned to the treasury.
+		type TreasuryAccount: Get<Self::AccountId>;
+
+		/// Origin from which any referendum may be cancelled in an emergency.
+		type CancellationOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Origin from which proposals may be blacklisted.
+		type BlacklistOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Origin from which a proposal may be cancelled and its backers slashed.
+		type CancelProposalOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Origin for anyone able to veto proposals.
+		///
+		/// # Warning
+		///
+		/// The number of Vetoers for a proposal must be small, extrinsics are weighted according to
+		/// [MAX_VETOERS](./const.MAX_VETOERS.html)
+		type VetoOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+		/// Period in blocks where an external proposal may not be re-submitted after being vetoed.
+		#[pallet::constant]
+		type CooloffPeriod: Get<Self::BlockNumber>;
+
+		/// The amount of balance that must be deposited per byte of preimage stored.
+		#[pallet::constant]
+		type PreimageByteDeposit: Get<BalanceOf<Self>>;
+
+		/// An origin that can provide a preimage using operational extrinsics.
+		type OperationalPreimageOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+		/// The Scheduler.
+		type Scheduler: ScheduleNamed<Self::BlockNumber, Self::Proposal, Self::PalletsOrigin>;
+
+		/// Overarching type of all pallets origins.
+		type PalletsOrigin: From<frame_system::RawOrigin<Self::AccountId>>;
+
+		/// The maximum number of votes for an account.
+		///
+		/// Also used to compute weight, an overly big value can
+		/// lead to extrinsic with very big weight: see `delegate` for instance.
+		#[pallet::constant]
+		type MaxVotes: Get<u32>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// The maximum number of public proposals that can exist at any time.
+		#[pallet::constant]
+		type MaxProposals: Get<u32>;
+	}
+
+	// TODO: Refactor public proposal queue into its own pallet.
+	// https://github.com/paritytech/substrate/issues/5322
+	/// The number of (public) proposals that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn public_prop_count)]
+	pub type PublicPropCount<T> = StorageValue<_, PropIndex, ValueQuery>;
+
+	/// The public proposals. Unsorted. The second item is the proposal's hash.
+	#[pallet::storage]
+	#[pallet::getter(fn public_props)]
+	pub type PublicProps<T: Config> = StorageValue<
+		_,
+		Vec<(PropIndex, ProposalId<T::Hash, T::AssetId>, T::AccountId)>,
```

Comment:
in what part of code ProposalId<T::Hash, T::AssetId>, uses as stucture with fields?

Reply:
can we have alias for it to ease review with original democrasy diff?